napari/layers/base/base.py

```diff
@@ -905,15 +913,47 @@ def help(self, help_text):
 
     @property
     def interactive(self):
-        """bool: Determine if canvas pan/zoom interactivity is enabled."""
-        return self._interactive
+        return self.mouse_pan or self.mouse_zoom
 
     @interactive.setter
     def interactive(self, interactive):
-        if interactive == self._interactive:
+        with self.events.interactive.blocker():
```

Comment:
```suggestion
        warnings.warn(trans._("Layer.interactive is deprecated since napari 0.5.0 and will be removed in 0.6.0. Please use Layer.mouse_pan and Layer.mouse_zoom instead"), FutureWarning, stacklevel=2)
        with self.events.interactive.blocker():
```